collectington/collectington_api.py

```diff
@@ -204,17 +204,20 @@ def call_prometheus_metrics(self, service_metric_dict, list_of_metric_instances)
         """
         for p_instance in list_of_metric_instances:
             metric = str(p_instance).split(":")[1]
-
+            labels_object = None
             if (
-                self.config.get("prometheus_metric_labels") is not None
-                and metric in self.config["prometheus_metric_labels"]
+                self.config["services"][self.service_name].get("prometheus_metric_labels") is not None
+                and metric in self.config["services"][self.service_name]["prometheus_metric_labels"]
             ):
+                labels_object = self.config["services"][self.service_name]["prometheus_metric_labels"]
+                break
+
+            if (labels_object is not None):
```

Comment:
line 207 and these lines can be removed as we are already checking this condition from line 209 (duplicate condition)  - `self.config["services"][self.service_name].get("prometheus_metric_labels") is not None`

Reply:
Ok, I'll look into refactoring it.

Reply:
Done, got rid of the unnecessary intermediate variable as you suggested.